ofproto/ofproto-dpif-xlate.c

```diff
@@ -3263,6 +3290,29 @@ xlate_normal(struct xlate_ctx *ctx)
         }
         p4ovs_unlock(&p4ovs_fdb_entry_lock);
     }
+
+    // Update the recently added MAC entry with flow info
+    struct mac_entry *e;
+
+    ovs_rwlock_wrlock(&ctx->xbridge->ml->rwlock);
+    e = mac_learning_lookup(ctx->xbridge->ml, flow->dl_src, vlan);
+    if (e) {
+        e->nw_src = flow->nw_src;
+        e->nw_dst = flow->nw_dst;
+     //TODO: Update IPv6 info in MAC entry when IPv6 support is added
+     }
+     ovs_rwlock_unlock(&ctx->xbridge->ml->rwlock);
+
+    if (ovs_p4_offload_enabled()) {
+        struct ip_mac_map_info ip_info;
+        memset(&ip_info, 0, sizeof(ip_info));
+        if(update_ip_mac_map_info(flow, &ip_info)) {
```

Comment:
```suggestion
        if (update_ip_mac_map_info(flow, &ip_info)) {
```
`if` is a keyword, not a function.